neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,438 @@
+
+        (
+            self._global_info,
+            self._ordered_channels,
+            data_dtype,
+            self._block_size,
+        ) = read_rhd(header_file)
+
+        self._raw_data = {}
+        for stream_index, sub_datatype in data_dtype.items():
+            if one_file_per_signal:
+                self._raw_data[stream_index] = np.memmap(
+                    raw_file_dict[stream_index], dtype=sub_datatype, mode="r"
+                )
+            else:
+                self._raw_data[stream_index] = []
+                for channel_index, datatype in enumerate(sub_datatype):
+                    self._raw_data[stream_index].append(
+                        np.memmap(
+                            raw_file_dict[stream_index][channel_index],
+                            dtype=[datatype],
+                            mode="r",
+                        )
+                    )
+
+        # check timestamp continuity
+        if one_file_per_signal:
+            timestamp = self._raw_data[6]["timestamp"].flatten()
+        else:
+            timestamp = self._raw_data[6][0]["timestamp"].flatten()
+        assert np.all(np.diff(timestamp) == 1), "timestamp have gaps"
+
+        # signals
+        signal_channels = []
+        for c, chan_info in enumerate(self._ordered_channels):
+            name = chan_info["native_channel_name"]
+            chan_id = str(c)  # the chan_id have no meaning in intan
+            if chan_info["signal_type"] == 20:
+                # exception for temperature
+                sig_dtype = "int16"
+            else:
+                sig_dtype = "uint16"
+            stream_id = str(chan_info["signal_type"])
+            signal_channels.append(
+                (
+                    name,
+                    chan_id,
+                    chan_info["sampling_rate"],
+                    sig_dtype,
+                    chan_info["units"],
+                    chan_info["gain"],
+                    chan_info["offset"],
+                    stream_id,
+                )
+            )
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+
+        stream_ids = np.unique(signal_channels["stream_id"])
+        signal_streams = np.zeros(stream_ids.size, dtype=_signal_stream_dtype)
+        signal_streams["id"] = stream_ids
+        for stream_index, stream_id in enumerate(stream_ids):
+            signal_streams["name"][stream_index] = stream_type_to_name.get(
+                int(stream_id), ""
+            )
+
+        self._max_sampling_rate = np.max(signal_channels["sampling_rate"])
+
+        if one_file_per_signal:
+            self._max_sigs_length = max(
+                [
+                    raw_data.size * self._block_size
+                    for raw_data in self._raw_data.values()
+                ]
+            )
+        else:
+            self._max_sigs_length = max(
+                [
+                    len(raw_data) * raw_data[0].size * self._block_size
```

Comment:
I was testing your files on gin and this is failing today. The reason being is that some of your raw_data items are still list at this stage (and therefore don't have the size attribute):

https://github.com/zm711/python-neo/blob/7b6af2f2f7969a6ac486aad5e5387770149f67a8/neo/rawio/intanbinaryrawio.py#L102-L103

I am still looking at your code so I am not sure about the best suggestion here.

Reply:
Okay, when I ran the test on a different dataset this worked. So I'll try to figure that out.

Reply:
Okay, so I tested this locally and it didn't fail? I'm not sure what exactly is failing.

Reply:
I figured it out. So the problem was I originally included a test file with all possible streams. But people do not always provide all possible streams. So my second dataset was missing the supply voltage stream. It turns out that the code required all streams. So in the line of code you flagged the issue was the supply voltage raw data was an empty list causing it to fail. I've fixed that issue by cleaning up the dtype but it wasn't very elegant. I'll flag it for you to look over and hopefully you'll have a better idea how to fix :)

Reply:
I think this is good. I feel the code should be robust to the absence of a stream. The files are in gin now. I will make an attempt to fix this.

Reply:
If you really want to stick to Sam implementation you can just filter out the stuff in the dictionary that are empty lists.

Reply:
I would just skip adding the streams whose ifles are not present. As in, skip blocks like this:

https://github.com/zm711/python-neo/blob/08c3ba46ee9d623eee5c8049f399fbb537677823/neo/rawio/intanbinaryrawio.py#L414-L427

If the corresponding files are not available.